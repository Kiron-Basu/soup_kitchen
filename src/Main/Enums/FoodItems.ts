export enum FoodItem {
  bread = "bread",
  cheese = "cheese",
  milk = "milk",
  butter = "butter",
  soup = "soup",
}
