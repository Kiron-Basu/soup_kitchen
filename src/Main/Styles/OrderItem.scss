.orderItem {
  display: flex;
  align-items: center;

  &__deleteIcon {
    margin-left: 5px;
  }

  &__description {
    text-align: start;
  }
}
