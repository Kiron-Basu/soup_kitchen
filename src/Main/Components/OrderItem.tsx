import React, { RefObject } from "react";
import Divider from "@material-ui/core/Divider";
import DeleteIcon from "@material-ui/icons/Delete";

import { poundify } from "../Utilities/InvoiceUtilities";
import { FoodItem } from "../Enums/FoodItems";
import "../Styles/OrderItem.scss";


interface IOrderItem {
  foodItem: string;
  quantity: number;
  subtotal: number;
  handleClearItem: (item: FoodItem) => void;
  ref: RefObject<SVGSVGElement>;
}
//@ts-ignore for now
export const OrderItem: React.FC<IOrderItem> = React.forwardRef(
  (props, ref): JSX.Element => {
    const { foodItem, quantity, subtotal, handleClearItem } = props;

    return (
      <div className="orderItem">
        <p className="orderItem__description">
          {foodItem} x {quantity} &nbsp; &nbsp; | &nbsp; &nbsp;{" "}
          {poundify(subtotal)}
        </p>
        <DeleteIcon
          ref={ref}
          className="orderItem__deleteIcon"
          color="secondary"
          onClick={() =>
            handleClearItem(FoodItem[foodItem as keyof typeof FoodItem])
          }
        />
        <Divider variant="fullWidth" />
      </div>
    );
  }
);
