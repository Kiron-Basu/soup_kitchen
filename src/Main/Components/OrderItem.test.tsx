import React from "react";
import { ShallowWrapper, shallow } from "enzyme";

import { OrderItem } from "./OrderItem";
import { FoodItem } from "../Enums/FoodItems";

describe("OrderItem tests", () => {
  const handleClearItem: jest.Mock = jest.fn();

  test("Renders OrderItem correctly", () => {
    const component: ShallowWrapper = shallow(
      <OrderItem
        quantity={0}
        subtotal={0}
        foodItem={FoodItem.bread}
        handleClearItem={handleClearItem}
      />
    );

    expect(component).toMatchSnapshot();
  });
});
