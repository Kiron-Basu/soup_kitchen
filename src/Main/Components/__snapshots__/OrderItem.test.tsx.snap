// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OrderItem tests Renders OrderItem correctly 1`] = `
<div
  className="orderItem"
>
  <p
    className="orderItem__description"
  >
    bread
     x 
    0
         |    
     
    £0.00
  </p>
  <Memo(ForwardRef(DeleteIcon))
    className="orderItem__deleteIcon"
    color="secondary"
    onClick={[Function]}
  />
  <WithStyles(ForwardRef(Divider))
    variant="fullWidth"
  />
</div>
`;
